### DISCUSSION TITLE
How to convert a YALMIP sdpvar into a normal variable

### DISCUSSION THREAD

**Discussion by khalid168** [yalmip/YALMIP#1335] [Optimization modelling] [answered]
I am building code for MPC and the integration/discretization method I am using is Backward Euler. To solve the implicit equation, I am using MATLAB's fsolve function. However, it is giving me this error. Fsolve requires following inputs to be of data type double : "X0''. As X0 is being passed as a variable from YALMIP, I guess its not being able to read the inner contents. How to get the data out of an sdpvar variable. Any other clues for why this error is coming are also welcome. Thanks a lot!

**Comment by johanlofberg** [comment, marked as answer]
It sounds like you are using the wrong tool. Incorporating a black-box function such as fsolve in a YALMIP model will not work

**Reply by khalid168** [reply]
Thanks for the response Johan! Can you please comment on why it will not work as doing just model discretization using my model and the backward Euler scheme works perfectly. It is just when I start doing it in YALMIP for MPC that it gives this error. 
Thank you!

**Comment by johanlofberg** [comment]
fsolve is a back-box which has various programming constructs,.i.e it is not just a mathematical operator, which is what YALMIP models

**Reply by khalid168** [reply]
Ok thanks a lot!
